### PR TITLE
8312190: Fix c++11-narrowing warnings in hotspot code

### PR DESCRIPTION
**Pull request by djelinski**
This patch fixes compilation warnings produced by Clang when compiling on Windows.

Clang emulates MSVC behavior and uses `int` for enumeration types that do not explicitly specify the underlying type. This patch sets an explicit underlying type for 3 enumerations to fix the warnings.

See Microsoft's documentation of [Zc:enumTypes](https://learn.microsoft.com/en-us/cpp/build/reference/zc-enumtypes?view=msvc-170) for more information.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312190](https://bugs.openjdk.org/browse/JDK-8312190): Fix c++11-narrowing warnings in hotspot code (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14907/head:pull/14907` \
`$ git checkout pull/14907`

Update a local copy of the PR: \
`$ git checkout pull/14907` \
`$ git pull https://git.openjdk.org/jdk.git pull/14907/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14907`

View PR using the GUI difftool: \
`$ git pr show -t 14907`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14907.diff">https://git.openjdk.org/jdk/pull/14907.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14907#issuecomment-1639427545)

### CODE REVIEW

File: src/hotspot/share/utilities/debug.hpp [hunk 1]
```diff
@@ -245,7 +245,7 @@ do {
 
 
 // types of VM error - originally in vmError.hpp
-enum VMErrorType {
+enum VMErrorType : unsigned int {
```
Comment (dholmes-ora) [inline src/hotspot/share/utilities/debug.hpp:248]: Why 'unsigned int' when the other changes are `uint`?

**Review by dholmes-ora — APPROVED** [review]
Seems reasonable.

Thanks.

**Comment by dholmes-ora** [comment]
Of course please ensure all our CI builds are successful too!

File: src/hotspot/share/utilities/debug.hpp (hunk 1 above)
Comment (djelinski) [reply src/hotspot/share/utilities/debug.hpp:248]: `uint` is not available in this header file, and I didn't want to add includes.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by djelinski** [comment]
Mach5 came back clean. Thanks for the reviews!

/integrate

**Comment by coleenp** [comment]
@djelinski Please review PR https://github.com/openjdk/jdk/pull/15056 and/or test with your compiler, please.

**Pull request closed without merging**